sensor/common/pubsub/lane/default_test.go

```diff
@@ -0,0 +1,267 @@
+		assert.NotNil(s.T(), lane)
+		defer lane.Stop()
+		laneImpl, ok := lane.(*defaultLane)
+		require.True(s.T(), ok)
+		assert.Equal(s.T(), 0, cap(laneImpl.ch))
+	})
+	s.Run("with default lane size", func() {
+		laneSize := 10
+		config := NewDefaultLane(pubsub.DefaultLane, WithDefaultLaneSize(laneSize))
+		assert.Equal(s.T(), pubsub.DefaultLane, config.LaneID())
+		lane := config.NewLane()
+		assert.NotNil(s.T(), lane)
+		defer lane.Stop()
+		laneImpl, ok := lane.(*defaultLane)
+		require.True(s.T(), ok)
+		assert.Equal(s.T(), laneSize, cap(laneImpl.ch))
+	})
+	s.Run("with negative lane size", func() {
+		laneSize := -1
+		config := NewDefaultLane(pubsub.DefaultLane, WithDefaultLaneSize(laneSize))
+		assert.Equal(s.T(), pubsub.DefaultLane, config.LaneID())
+		lane := config.NewLane()
+		assert.NotNil(s.T(), lane)
+		defer lane.Stop()
+		laneImpl, ok := lane.(*defaultLane)
+		require.True(s.T(), ok)
+		assert.Equal(s.T(), 0, cap(laneImpl.ch))
+	})
+	s.Run("with custom consumer", func() {
+		config := NewDefaultLane(pubsub.DefaultLane, WithDefaultLaneConsumer(newTestConsumer))
+		assert.Equal(s.T(), pubsub.DefaultLane, config.LaneID())
+		lane := config.NewLane()
+		assert.NotNil(s.T(), lane)
+		defer lane.Stop()
+		laneImpl, ok := lane.(*defaultLane)
+		require.True(s.T(), ok)
+		assert.NotNil(s.T(), laneImpl.newConsumerFn)
+		assert.Len(s.T(), laneImpl.consumerOpts, 0)
+	})
+	s.Run("with custom consumer and consumer options", func() {
+		config := NewDefaultLane(pubsub.DefaultLane, WithDefaultLaneConsumer(newTestConsumer, func(_ pubsub.Consumer) {}))
+		assert.Equal(s.T(), pubsub.DefaultLane, config.LaneID())
+		lane := config.NewLane()
+		assert.NotNil(s.T(), lane)
+		defer lane.Stop()
+		laneImpl, ok := lane.(*defaultLane)
+		require.True(s.T(), ok)
+		assert.NotNil(s.T(), laneImpl.newConsumerFn)
+		assert.Len(s.T(), laneImpl.consumerOpts, 1)
+	})
+}
+
+type testLaneConfig struct {
+	opts []pubsub.LaneOption
+}
+
+type testLane struct{}
+
+func (t *testLane) Publish(_ pubsub.Event) error {
+	return nil
+}
+
+func (t *testLane) RegisterConsumer(_ pubsub.Topic, _ pubsub.EventCallback) error {
+	return nil
+}
+
+func (t *testLane) Stop() {
+}
+
+func (lc *testLaneConfig) NewLane() pubsub.Lane {
+	ret := &testLane{}
+	for _, opt := range lc.opts {
+		opt(ret)
+	}
+	return ret
+}
+func (lc *testLaneConfig) LaneID() pubsub.LaneID {
+	return pubsub.DefaultLane
+}
+
+func (s *defaultLaneSuite) TestOptionPanic() {
+	defer goleak.AssertNoGoroutineLeaks(s.T())
+	s.Run("panic if WithDefaultLaneSize is used in a different lane", func() {
+		config := &testLaneConfig{
+			opts: []pubsub.LaneOption{
+				WithDefaultLaneSize(10),
+			},
+		}
+		s.Assert().Panics(func() {
+			config.NewLane()
+		})
+	})
+	s.Run("panic if WithDefaultLaneConsumer is used in a different lane", func() {
+		config := &testLaneConfig{
+			opts: []pubsub.LaneOption{
+				WithDefaultLaneConsumer(nil),
+			},
+		}
+		s.Assert().Panics(func() {
+			config.NewLane()
+		})
+	})
```

Comment:
I comment on this because I saw that `WithDefaultLaneConsumer` can accept nils that would panic later when calling `RegisterConsumer` with nil `newConsumerFn`. I see you are testing a panic here, but not for `RegisterConsumer`.

I see two options here:
- If `nil` is a valid argument to `WithDefaultLaneConsumer`, then let's also add test case where we later call `RegisterConsumer` and assert that it panics.
- If not, then maybe let's disallow calling `WithDefaultLaneConsumer` with nil arg (and potentially fallback to the `NewDefaultConsumer`?)

Reply:
I went with a third option of panicking in the `WithDefaultLaneConsumer` if a `nil` consumer is set. I think it doesn't make sense to have a `nil` consumer and it would be better to panic in the `WithDefaultLaneConsumer` rather than `RegisterConsumer`. WDYT?